### PR TITLE
🎉 SDL1: An option to render directly to video mem

### PR DESCRIPTION
**Pull request by glebm**
Now that we no longer have a buffer border, we can render directly to
the video memory if the output surface is 8-bit and double buffering is
enabled.

Also adds a flag to force this even when double buffering is disabled,
because some systems emulate 8-bit mode with something else, so the
output is always buffered.

This required a tweak to the cursor to make sure that we clear it after
`SDL_Flip`.

### CODE REVIEW

File: Source/scrollrt.cpp [hunk 1]
```diff
@@ -143,7 +143,7 @@ static void BlitCursor(BYTE *dst, int dst_pitch, BYTE *src, int src_pitch)
 /**
  * @brief Remove the cursor from the buffer
  */
-static void scrollrt_draw_cursor_back_buffer(const CelOutputBuffer &out)
+static void ClearCursor(const CelOutputBuffer &out)
```
Comment (AJenbo) [inline Source/scrollrt.cpp:146]: :+1:

File: Source/scrollrt.cpp [hunk 2]
```diff
@@ -1550,18 +1551,19 @@ void DrawAndBlit()
...
 
 	DrawMain(hgt, ddsdesc, drawhpflag, drawmanaflag, drawsbarflag, drawbtnflag);
 
+	RenderPresent();
+
 	lock_buf(0);
-	scrollrt_draw_cursor_back_buffer(GlobalBackBuffer());
+	ClearCursor(GlobalBackBuffer());
 	unlock_buf(0);
-	RenderPresent();
```
Comment (AJenbo) [inline Source/scrollrt.cpp:1564]: Did you test this both on SDL1 and SDL2?

Comment (glebm) [reply Source/scrollrt.cpp:1564]: Yeah, on SDL2 we clear the cursor from `pal_surface` (which is always a separate surface there) so the order doesn't matter

Comment (glebm) [reply Source/scrollrt.cpp:1564]: One issue is that if we're double-buffering in direct rendering mode, we're unable to clear the cursor from the back buffer -- but we can look into this later as the direct rendering feature is still experimental

File: Source/scrollrt.cpp (hunk 1 above)
Comment (glebm) [reply Source/scrollrt.cpp:146]: Oops, there is already a ClearCursor

File: Source/scrollrt.cpp (hunk 2 above)
Comment (glebm) [reply Source/scrollrt.cpp:1564]: Fixed

File: Source/scrollrt.cpp (hunk 1 above)
Comment (glebm) [reply Source/scrollrt.cpp]: Fixed

Comment (AJenbo) [reply Source/scrollrt.cpp]: that's good as well :)

**Pull request merged**